modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/views/table/dnd_table/ContextProvider.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import React, {useCallback, useMemo, useState} from 'react';
+
+import Context from './TableContext';
+
+function ContextProvider({children, columnsNames}) {
+	const [tableWidth, updateTableWidth] = useState(null);
+	const [columnsDefinitions, updateColumnsDefinitions] = useState(new Map());
+	const [draggingColumnName, updateDraggingColumnName] = useState(null);
+	const [draggingAllowed, updateDraggingAllowed] = useState(true);
+
+	const registerColumn = (name, width, resizable) => {
+		updateColumnsDefinitions((definitions) => {
+			const updatedDefinitions = new Map(definitions);
+
+			updatedDefinitions.set(name, {
+				resizable,
+				width,
+			});
+
+			return updatedDefinitions;
+		});
+	};
+
+	const isFixed = useMemo(() => {
+		const columnsAreAllRegistered = columnsNames.reduce(
+			(registered, name) => {
+				return registered && !!columnsDefinitions.get(name);
+			},
+			true
+		);
```

Comment:
`every()` is probably easier to read than `reduce()`:

```js
const allRegistered = columnNames.every((name) => }
    return columnDefinitions.has(name);
});
```

Has the added benefit of short-circuiting as soon as it sees the first unregistered name.

Also note the use of `has()` instead of `get()`. Generally it's better to use a more specific method when one is available, because it makes your intent clearer (reader doesn't have to think, "how is this value used?"; they just see it and think, "this code tests for the presence of a value").